### PR TITLE
Make NT Default possible roundstart lawset (no meme laws edition)

### PR DESCRIPTION
**Pull request by moxian**
## What Does This PR Do
This is an alternative to #15157

- Refactors the way lawsets are set as "eligible to be roundstart" - it is now encoded in law selection logic, rather than on the law itself; Also made the eligibility weighted, rather than a yes/no.
- NT Default is now roundstart default, along with what we had previously in Crewsimov and Corporate
- ~~Added a small chance for meme laws (T.Y.R.A.N.T, Robocop, PALADIN, NT Agressive) to appear roundstart as well.~~

## Why It's Good For The Game
NT Default is an overall decent lawset that command seems to gravitate to roundstart anyway, so there's no reason not to have it in rotation.
~~The meme laws are there to add a bit of spice to the AIs lives. With a very low probability, of course.~~

## Images of changes
dreamseeker-20201220-163115

## Changelog
:cl:
add: NT Default is now roundstart lawset along with Corporate and Crewsimov
/:cl:

### CODE REVIEW

**Comment by kenken244** [comment]
TYRANT should probably never ever appear, I can see NT experimenting with the others (this is a science station after all) but tyrant is just so bad it would prevent the station from functioning. and also invite the ai to murder people. I do like the idea of having a small chance of a very different lawset, but I think it would end up just being changed as soon as anyone realized what laws the ai is on

**Comment by moxian** [comment]
> TYRANT should probably never ever appear

Happy to do this change if a second person (preferably, game admin, but not essential) agrees. I don't have a strong opinion of my own here.

> also, maybe have a chance of the ai being completely unlawed instead?

I'm not sure how it's any better than TYRANT.
(And also it would require a bit more refactoring, so I'd like to find an excuse not to :P)

**Comment by Marginalorb** [comment]
T.Y.R.A.N.T. doesn't really invite unwarranted violence, in fact, it tells the AI to respect authority and to act with discipline. Granted, it does open up the possibility of the AI assisting someone in mutiny, but again, the laws specify the AI should punish those who disrespect authority, unless they're more fit to hold said authority.

Honestly I wouldn't mind seeing a higher chance for the other laws to come up, but y'know, I'm just a GA. Might want to see what a HA or Maint think before changing anything.

**Comment by kenken244** [comment]
tyrant pretty much blatantly encourages the ai to kill command and take control of the station if it thinks that it can do so.

**Comment by Spacemanspark** [comment]
robocop and paladin are a bit much. They were actually removed from the rotation years back. They're very validhunty lawsets, fun in small doses and in certain situations... but not something to see on the daily. 

Tyrant... I don't know, bit iffy on it.

@kenken244 is wrong, however. It doesn't encourage an AI to murder command whatsoever, unless they decide to attack it first.

It does often end in an AI being at odds with at least a few command members, however. I'm not sure if it's a good idea.

**Comment by procdrone** [comment]
I personally like the novelty of it. Command would probably quickly change the laws anyway if they did not find them to their liking. Having an unique thing to work with so rarely propably wouldn't be TOO much. + with this change almost everyone would question AI about their lawset immediately, given it can no longer be a cresimov/corpo only

**Comment by moxian** [comment]
> They're very <..> fun in small doses <..> but not something to see on the daily

This is exactly the intent of this PR - introduce those lawsets in small doses.
Please, again, note that even if they *happen* once a day, you won't *see* them daily because you're not online 24/7.

It is fair to view the lawsets as "antaggy", and basically treat as a midround-lite. I just think that, essentially, midround that happens at the very start of the round is not a bad thing (especially if they don't require blob levels of station response).

If the proposed frequency is seen as too high, I'm more than happy to adjust it - just give me an idea what numbers you'd like to see.

**Review by Kyep — COMMENTED** [review]
I'm fine with adding NT Default to the possible laws at roundstart.
I'm not a fan of having any chance to load something like Tyrant or Robocop.

**Comment by Spacemanspark** [comment]
I'd prefer robocop/ paladin were left as lawsets for the crew themselves to add

File: code/modules/mob/living/silicon/laws.dm [hunk 1]
```diff
@@ -125,18 +125,25 @@
...
-	for(var/law in paths)
-		var/datum/ai_laws/L = new law
-		if(!L.default)
-			continue
-		law_options += L
-	return pick(law_options)
+	// The weights here add up to ~900. That is 10x the average number of rounds we have in a week.
+	// So a law that has a weight of 10 would appear ~once a week.
+	// (At the time of writing, the average round length in the preceeding 3 months is ~113 minutes)
+	var/weights = list(
```
Comment (farie82) [inline code/modules/mob/living/silicon/laws.dm:137]:
```suggestion
	var/static/list/weights = list(
```

**Review by farie82 — CHANGES_REQUESTED** [review]

**Comment by moxian** [comment]
Chatted with Kyet, gained a new perspective, and I now agree that having "meme" laws roundstart is probably not a good idea. (Or rather, I still think it *might* be a good idea, but it needs a completely different framing, and much more thorough justification than just "it might be fun, idk").
Either way, this part is now removed from the PR.

This now has the exactly gameplay impact as #15157 , just a slightly different implementation, that would allow to easily tweak lawset probabilities in the future (if need be).

**Review by farie82 — APPROVED** [review]

**Review by Dumbdumn5 — APPROVED** [review]
Considering the changes made and the roundstart list for AI lawsets including more or less the blandest of the bunch (which isn't necessarily a bad thing), consider this my approval, g o b l i n :+1:

**Comment by AffectedArc07** [comment]
Closed in favour of #15157

**Pull request closed without merging**